webapp/models.py

```diff
@@ -37,62 +49,76 @@ def __init__(self, base_url, frontpage_id, session_class=CachedSession):
 
         self.base_url = base_url.rstrip("/")
         self.frontpage_id = frontpage_id
-        self.session = CachedSession(expire_after=300)
+        self.session = session
+
+    def __del__(self):
+        self.session.close()
 
-    def get_topic(self, path):
+    def get_document(self, path):
         """
-        Retrieve topic object by path
+        Retrieve and return relevant data about a document:
+        - Title
+        - HTML content
+        - Navigation content
         """
 
-        response = self.session.get(
-            f"{self.base_url}/t/{path}.json", allow_redirects=False
-        )
-        response.raise_for_status()
+        parse_error = None
 
-        if response.status_code >= 300:
-            raise RedirectFoundError(response=response)
+        try:
+            frontpage, nav_html = self.parse_frontpage()
+        except NavigationParseError as err:
```

Comment:
This is not tested.

Reply:
It is - here: https://github.com/canonical-websites/discourse-docs.snapcraft.io/pull/34/files#diff-0f3b2e8da7eaa031d74d097803cf786aR157